Source/Scene/Scene.js

```diff
@@ -1651,115 +1659,128 @@ define([
-                    vertexFormat : PerInstanceColorAppearance.FLAT_VERTEX_FORMAT
-                })));
-
-                scene._debugVolume = new Primitive({
-                    geometryInstances : new GeometryInstance({
-                        geometry : geometry,
-                        modelMatrix : Matrix4.fromRotationTranslation(halfAxes, center, new Matrix4()),
-                        attributes : {
-                            color : new ColorGeometryInstanceAttribute(1.0, 0.0, 0.0, 1.0)
-                        }
-                    }),
-                    appearance : new PerInstanceColorAppearance({
-                        flat : true,
-                        translucent : false
-                    }),
-                    asynchronous : false
-                });
-            }
+        if ((defined(scene.debugCommandFilter)) && !scene.debugCommandFilter(command)) {
+            return;
+        }
+
+        if (command instanceof ClearCommand) {
+            command.execute(context, passState);
+            return;
+        }
 
-            var savedCommandList = frameState.commandList;
-            var commandList = frameState.commandList = [];
-            scene._debugVolume.update(frameState);
+        if (command.debugShowBoundingVolume && (defined(command.boundingVolume))) {
+            debugShowBoundingVolume(command, scene, passState, debugFramebuffer);
+        }
 
-            var framebuffer;
-            if (defined(debugFramebuffer)) {
-                framebuffer = passState.framebuffer;
-                passState.framebuffer = debugFramebuffer;
-            }
+        if (scene.debugShowCommands || scene.debugShowFrustums) {
+            executeDebugCommand(command, scene, passState);
+            return;
+        }
 
-            commandList[0].execute(context, passState);
+        if (frameState.passes.depth && defined(command.derivedCommands.depth)) {
+            command.derivedCommands.depth.depthOnlyCommand.execute(context, passState);
+            return;
+        }
 
-            if (defined(framebuffer)) {
-                passState.framebuffer = framebuffer;
-            }
+        var shadowsEnabled = scene.frameState.shadowHints.shadowsEnabled;
+        var lightShadowsEnabled = shadowsEnabled && (scene.frameState.shadowHints.lightShadowMaps.length > 0);
 
-            frameState.commandList = savedCommandList;
+        if (lightShadowsEnabled && command.receiveShadows && defined(command.derivedCommands.shadows)) {
+            // If the command receives shadows, execute the derived shadows command.
+            // Some commands, such as OIT derived commands, do not have derived shadow commands themselves
+            // and instead shadowing is built-in. In this case execute the command regularly below.
+            command.derivedCommands.shadows.receiveCommand.execute(context, passState);
+        } else {
+            command.execute(context, passState);
```

Comment:
All these changes are just reorganization without code edits. The debug show bounding volume code is just moved to its own function.